torch/csrc/jit/tensorexpr/kernel.h

```diff
@@ -101,7 +101,8 @@ class TORCH_API TensorExprKernel {
       const torch::jit::Value* v,
       const std::function<
           ExprHandle(const ExprHandle&, const ExprHandle&, const ExprHandle&)>&
-          innerExpr);
+          innerExpr,
+      bool promote_inputs = true);
```

Comment:
Minor: it doesn't feel that "promote_inputs" is true/false is two flavors of this function. My first impression is that "promote_inputs = true" is the public behavior, and "promote_inputs = false" is an internal behavior. I would prefer to remove this argument from the public function, and refactor the "= false" flavor to a "_interna"/"_impl" function.